lld/test/COFF/lto-cache-errors.ll

```diff
@@ -1,15 +1,26 @@
-; REQUIRES: x86, non-root-user
-;; Not supported on windows since we use permissions to deny the creation
-; UNSUPPORTED: system-windows
+;; The output is OS-specific, so this test is limited to Windows.
```

Comment:
Switching a test from "not windows" to "only windows" is suspicious. Is it just the error message phrasing? Can you follow the pattern of other tests that depend on error strings, e.g. llvm/test/tools/llvm-ar/error-opening-permission.test

Reply:
Thanks. I made the test Windows specific because there is additional output on Windows (that I'm trying to eliminate in this patch) so that seems a better platform to run this test on than Linux. It's stell testing the same failure condition. I'll try to either make the test work on both platforms or add a separate Windows specific test.

Reply:
I have had to change the test strategy for this to use the invalid pass pipeline errors, as they occur in a single-threaded context*. The test for that in LLD works on both platforms.

* #140955 was reverted. Digging into why the test failed, I suspect that reportFatalUsageError can cause a crash if threads call reportFatalUsageError simultaneously (the ThinLTO cache dir is used from a multi-threaded context). I have observed this on Windows. It may also apply on Linux.